earth_enterprise/src/fusion/autoingest/StorageManager.h

```diff
@@ -355,5 +368,30 @@ bool StorageManager<AssetType>::SaveDirtyToDotNew(
   return true;
 }
 
+template<class AssetType>
+bool StorageManager<AssetType>::DetermineIfPrune() {
+  std::lock_guard<std::recursive_mutex> lock(storageMutex);
+  uint32 cacheSize = CacheSize();
+  uint32 dirtySize = DirtySize();
+  uint32 oriNum;
+  uint32 newNum;
+  uint32 diff;
+  float percent;
+
+  if (cacheSize > dirtySize) {
+    oriNum = cacheSize;
+    newNum = dirtySize;
+  }
+  else {
+    oriNum = dirtySize;
+    newNum = cacheSize;
+  }
+  if (oriNum == newNum || oriNum == 0 || newNum == 0) { return false; }
```

Comment:
I would return `true` if either of the sizes is 0. If the cache size is 0, it's small enough that it doesn't matter because it won't take long to prune anyway. If the dirty map size is zero, there's no downside to pruning the cache because the dirty map isn't forcing a bunch of stuff to stay in the cache. However, if the two are equal we don't want to prune because it will be wasted effort. However, I think that case will be caught by your calculations below.